app/firebase/reservations/reservation.ts

```diff
@@ -0,0 +1,38 @@
+import { getFirestore, doc, getDoc } from 'firebase/firestore';
+import { ReservationType } from '@/app/components/Reservations/Reservation';
+import firebaseApp from '../config';
+
+const reservationId = 'ioPMkgCdwfB8xfsYd9Ze';
+
+export async function getReservation(): Promise<ReservationType> {
+	const db = getFirestore(firebaseApp);
+	const docRef = doc(
+		db,
+		process.env.NEXT_PUBLIC_RESERVATIONS_COLLECTION,
+		reservationId
+	);
+	const docSnap = await getDoc(docRef);
+	if (!docSnap.exists()) {
+		throw new Error('The reservation does not exist.');
+	}
+	const data = { ...docSnap.data() };
+	return {
+		court: data.court,
+		owner: data.owner,
+		duration: data.duration,
```

Comment:
The firebase service should be in charge of calculate the `finalTime` based on `initialTime` and `duration` fields.

You can use `moment` library, which was already added to the project.

Here is an example of how you can implement it:
```suggestion
	finalTime: moment(data.initialTime.toDate()).add(data.duration, 'minutes').toDate(),	
```

Reply:
remember to rebase to origin/master to get access to the moment library 😄

Reply:
I'm wondering if this should be part of the model or just the representation in the UI Component?

Reply:
I agree with you. For now we can leave it as it is to move forward with the development. We will review the current design later.